docs: include call chain new feature (#12)
r0gue-io/pop-docs: pop-cli-for-appchains/guides/calling-a-chain.md

# Call a chain

### Interactive Guidance (Recommended)

> Interact with the chain **with** Pop CLI's interactive guidance by simply entering:

Reviewer: ```suggestion
Interact with a chain **using** Pop CLI's interactive guidance by simply entering:
```

Author: https://github.com/r0gue-io/pop-docs/pull/12/commits/1090c54a67f8a58d84e3b649f99877b70aa9aeaa

```shell
pop call chain
```

> You will be prompted to select a pallet, choose a dispatchable function, specify any required arguments, then provide connection details and the information of the account that will sign the extrinsic.

Reviewer 2: We should be careful with using extrinsic in stead of transaction, the latter is easier to understand for most people.
```suggestion
You will be prompted to select a pallet, the dispatchable function and its required arguments, connection details for the chain, and the account to sign the transaction.
```

Author: https://github.com/r0gue-io/pop-docs/pull/12/commits/0447185c620cce672862cfcdc515396c75e85671

<figure><img src="../.gitbook/assets/callchain.gif" alt="pop call chain"><figcaption><p>pop call chain</p></figcaption></figure>

### Manual (non-interactive)

> If you prefer not to use interactive prompts, you can call the chain by specifying all the required arguments directly. For example, to call a dispatchable function on a specific pallet:

Reviewer 2: ```suggestion
If you prefer not to use interactive prompts, you can call the chain by specifying all the required arguments directly:
```

```shell

> pop call chain --pallet System --function remark --args "0x11" --url ws://localhost:9944/ --suri //Alice

Reviewer 2: This will still prompt for sudo and `skip-confirm` which means it is still interactive

Author: https://github.com/r0gue-io/pop-docs/pull/12/commits/0447185c620cce672862cfcdc515396c75e85671

```

```
┌   Pop CLI : Call a chain
│
◇  Would you like to dispatch this function call with `Root` origin?
│  No 
│
⚙  pop call chain --pallet System --function remark --args "0x11" --url ws://localhost:9944/ --suri //Alice
│  
⚙  Encoded call data: 0x00000411
│  
◇  Do you want to submit the extrinsic?
│  Yes 
│
◇  Extrinsic submitted with hash: "0xadbfbb2f92b22a2c5d6542ba80e4111b5c60057f594a6d155341879c5a46f96e"
│
└  Call complete.
```

#### Additional Options

> * If you have sudo rights and need to dispatch a function as Root origin you can specify it using the flag `--sudo`:

Reviewer: I believe it is OK if we use more polkadot-sdk vocabulary here.

Something like:

To dispatch a call with the Root origin when the chain's runtime includes `pallet-sudo`, you can wrap the call in a `sudo.sudo()` call by using the `--sudo` flag.

Author: Good point! I agree, your approach is a better fit for the docs.

> * To dispatch a call with Root origin when the chain's runtime includes `pallet-sudo`, you can wrap the call in a `sudo.sudo()` call by using the `--sudo` flag:

Reviewer 2: ```suggestion
* To dispatch a call with Root origin, provided the chain uses `pallet-sudo`, you can use the `--sudo` flag:
```

Author: I prefer the first one here, explaining that we wrap the call in a `sudo.sudo()` . It was suggested by Ale here https://github.com/r0gue-io/pop-docs/pull/12#discussion_r1878876219

Reviewer 2: I think it is too code specific but oke

> * To dispatch a call with the Root origin when the chain's runtime includes `pallet-sudo`, you can wrap the call in a `sudo.sudo()` call by using the `--sudo` flag:

Reviewer: ```suggestion
* To dispatch a call with Root origin when the chain's runtime includes `pallet-sudo`, you can wrap the call in a `sudo.sudo()` call by using the `--sudo` flag:
```

Author: Ahh makes more sense

```shell
pop call chain --pallet System --function remark --args "0x11" --url ws://localhost:9944/ --suri //Alice --sudo
```

> * If you already have the SCALE-encoded call data, and want to directly submits the extrinsic:

Reviewer 2: ```suggestion
* If you already have the SCALE-encoded call data and want to directly submit the extrinsic:
```
